js/interface.js

```diff
@@ -937,16 +941,26 @@ var DynamicLists = (function() {
 
       loadingPromise = new Promise(function(resolve) {
         if (!_this.config.dataSourceId) {
-          _this.updateFieldsWithColumns(_this.config.defaultColumns);
-          initDataSourceProvider(_this.config.dataSourceId);
+          _this.initDefaultDatasourceProvider();
 
           resolve();
         } else {
-          $document.on('datasource-selected', function() {
+          // Check data source
+          _this.getDataSourceById(_this.config.dataSourceId).then(function() {
+            $document.on('datasource-selected', function() {
+              resolve();
+            });
+
+            _this.initDataSourceProvider(_this.config.dataSourceId);
+
+            return;
```

Comment:
Why is this `return` necessary?

Reply:
> Why is this `return` necessary?

Apologies, this code was from previous implementation. I'll fix that